backend/tests/conftest.py

```diff
@@ -697,12 +706,28 @@ def sync_status(self) -> tuple[dict, int]:
         rv = self._flask_client.get("/info/sync-status")
         return json.loads(rv.text), rv.status_code
 
-    def wait_for_sync(self, target):
+    def wait_for_sync(self, target, timeout_s=20):
+        timeout = datetime.timedelta(seconds=timeout_s)
+        start = datetime.datetime.now()
         while True:
-            res, _ = self.sync_status()
-            if res["indexedEpoch"] == res["blockchainEpoch"]:
-                if res["indexedEpoch"] == target:
-                    return
+            try:
+                res, status_code = self.sync_status()
+                current_app.logger.debug(f"sync_status returns {res}")
+                current_app.logger.debug(
+                    f"sync_status http status code is {status_code}"
+                )
+                assert status_code == 200
+                if res["indexedEpoch"] == res["blockchainEpoch"]:
+                    if res["indexedEpoch"] == target:
+                        return
+            except Exception as exp:
```

Comment:
You can consider specifying exact Exception here, it'll be probably ```HTTPException```. Thanks to that we're prone that we'll not be supressing errors that come from the improper structure of the res variable.

Reply:
It is unclear what exceptions it throws and if it even throws them. I've refactored this code to make sure that case you've mentioned doesn't get captured.